Improve SC docs (#438)
aeternity/protocol: channels/state.md

it.

The means for achieving its goal of off-chain communication is reaching
consensus. As long as participants agree that something is right, it is right
for their state channel. In this sense - rules not only can be bent, they can
be broken and participants can store their state in whatever structure they
see fit.  Although this is not a problem for off-chain consensus, it imposes a
problem for using the on-chain as an arbiter in case of a dispute. In order
for participants to be safely using a channel, at any point of time they
should be able to unilaterally call a contract or close the channel. This
relies on on-chain being able to read and - in the case of force progress -
deterministically update the channel's off-chain state.

## Structure

The off-chain state has the exact same structure as the on-chain state. As
described in [serialization document](/serializations.md#state-trees) the
off-chain state consists of subtrees. Those are:

* accounts - where accounts and contract balances are stored

* contracts - where the contracts and their states are stored

* calls - where the contract call from the last update are stored. This tree

>   is being purged at every round so the state of the channel is not increased

Reviewer: ```suggestion
  is being purged at every round so the state of the channel is not enlarged
```

>   by calls. This is useful especially in long running channels with a lot of

Reviewer: ```suggestion
  by calls. This is important especially for long running channels with many
```

>   calls. This is really important for forced progress transaction as if the

Reviewer: ```suggestion
  calls. This has implications for forced progress transactions as if the
```
